rules/msk_topic_config.go

```diff
@@ -229,3 +235,58 @@ func (r *MSKTopicConfigRule) validateCompressionType(
 	}
 	return nil
 }
+
+const (
+	cleanupPolicyKey     = "cleanup.policy"
+	cleanupPolicyDefault = "delete"
+)
+
+var (
+	cleanupPolicyDefaultFix  = fmt.Sprintf(`"%s" = "%s"`, cleanupPolicyKey, cleanupPolicyDefault)
+	cleanupPolicyValidValues = []string{"delete", "compact"}
+)
+
+func (r *MSKTopicConfigRule) ValidateCleanupPolicy(
+	runner tflint.Runner,
+	config *hclext.Attribute,
+	configKeyToPairMap map[string]hcl.KeyValuePair,
+) error {
+	cpPair, hasCp := configKeyToPairMap[cleanupPolicyKey]
+	if !hasCp {
+		err := runner.EmitIssueWithFix(
+			r,
+			fmt.Sprintf("missing %s: using default '%s'", cleanupPolicyKey, cleanupPolicyDefault),
+			config.Range,
+			func(f tflint.Fixer) error {
+				return f.InsertTextAfter(config.Expr.StartRange(), "\n"+cleanupPolicyDefaultFix)
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("emitting issue with fix: no cleanup policy: %w", err)
+		}
+		return nil
+	}
+
+	var cpVal string
+	diags := gohcl.DecodeExpression(cpPair.Value, nil, &cpVal)
```

Comment:
Why are we not passing context to decodeExpression? The [documentation](https://pkg.go.dev/github.com/hashicorp/hcl/v2/gohcl#DecodeExpression) states, that 

>The given EvalContext is used to resolve any variables or functions in expressions encountered while decoding. This may be nil to require only constant values, for simple applications that do not support variables or functions. 

But we have both [functions](https://github.com/utilitywarehouse/kafka-cluster-config/blob/f02723c25b6493d3a5f6bd419d33026cc0f086a9/modules/tls-app/main.tf#L12) and [variables](https://github.com/utilitywarehouse/kafka-cluster-config/blob/f02723c25b6493d3a5f6bd419d33026cc0f086a9/modules/tls-app/variables.tf)

Reply:
> But we have both [functions](https://github.com/utilitywarehouse/kafka-cluster-config/blob/f02723c25b6493d3a5f6bd419d33026cc0f086a9/modules/tls-app/main.tf#L12) and [variables](https://github.com/utilitywarehouse/kafka-cluster-config/blob/f02723c25b6493d3a5f6bd419d33026cc0f086a9/modules/tls-app/variables.tf)

We don't really expect functions or variables in the `kafka_topic` resources, e.g. we expect most of them to look like https://github.com/utilitywarehouse/kafka-cluster-config/blob/f02723c25b6493d3a5f6bd419d33026cc0f086a9/dev-aws/kafka-shared-msk/pubsub/pubsub-test-throughput.tf#L1

`tflint` doesn't really give you any support for handling variables either, making it a bit of a hassle to do, see https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/commit/6034af11858ea05654230a5a7c367d7b8b793ccf for some detail